kroxylicious-operator/src/test/java/io/kroxylicious/kubernetes/operator/LocallyRunningOperatorRbacHandler.java

```diff
@@ -0,0 +1,227 @@
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.fabric8.kubernetes.client.dsl.EditReplacePatchable;
+import io.fabric8.kubernetes.client.dsl.NonNamespaceOperation;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.javaoperatorsdk.operator.junit.AbstractOperatorExtension;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+import static io.kroxylicious.kubernetes.operator.ResourcesUtil.name;
+
+/**
+ * Designed to cooperate with {@link io.javaoperatorsdk.operator.junit.LocallyRunOperatorExtension}
+ * so that the operator runs locally with role based access restrictions as close to that it will
+ * have when deployed to Kubernetes for real.
+ * <br/>
+ * Register this extension before LocallyRunOperatorExtension, passing LocallyRunOperatorExtension
+ * the kubernetes client returned by {@link #operatorClient}.  This client will use a principal
+ * which has been configured in Kubernetes to have the same RBAC rights as the operator will use
+ * in production.
+ * <br/>
+ * For the test's interactions with Kubernetes, where more privileges are required, use the operations
+ * exposed by {@link #testActor(AbstractOperatorExtension)}.
+ *
+ */
+public class LocallyRunningOperatorRbacHandler implements BeforeEachCallback, AfterEachCallback {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LocallyRunningOperatorRbacHandler.class);
+
+    private final String impersonatedUser = UUID.randomUUID().toString();
+
+    @AutoClose
+    private final KubernetesClient operatorClient = OperatorTestUtils.kubeClient(
+            new KubernetesClientBuilder().editOrNewConfig().withImpersonateUsername(impersonatedUser).endConfig());
+
+    @AutoClose
+    private final KubernetesClient testActorClient = OperatorTestUtils.kubeClient();
+
+    // @formatter:off
+    private final ClusterRole frameworkClusterRole = new ClusterRoleBuilder()
+            .withNewMetadata()
+                .withName("framework-cluster-role")
+            .endMetadata()
+            .addNewRule()
+                .addToApiGroups("")
+                .addToVerbs("get", "create", "delete", "patch")
+                .addToResources("namespaces")
+            .endRule()
+            .addNewRule()
+                .addToApiGroups("apiextensions.k8s.io")
+                .addToVerbs("get", "list", "watch", "create", "delete", "patch", "delete")
+                .addToResources("customresourcedefinitions")
+                .endRule()
+            .build();
+    // @formatter:on
+
+    private final Path resourceDirectory;
+    private final List<PathMatcher> clusterRolePathMatchers;
+    private List<ClusterRole> clusterRoles;
+    private List<ClusterRoleBinding> roleBindings;
+
+    public LocallyRunningOperatorRbacHandler(Path resourceDirectory, String... clusterRoleFileGlobs) {
```

Comment:
nit: Is it worth adding an overload here that just takes a string for `resourceDirectory ` and does `Path.of(resourceDirectory)` given thats what each invocation of this does anyway?

Reply:
done